shapeshift/web
# feat: cosmos opportunities for defi earn

## Description

This enables cosmos opportunities on defi / earn page.

## Notice




## Pull Request Type


## Issue (if applicable)


https://github.com/shapeshift/web/issues/1402

## Risk


`<AssetCell>` component has been modified as well as `[OpportunityCard.tsx]` so existing behaviour for opportunity cards and defi/earn need to be regression tested

## Testing

- Connect with cosmos wallet
- Have some staked amount 
- Navigate to defi / earn
- Check validators with staked amount appear at top of list for all opportunities
- Check those opportunities also appear as cards
- Verify click behaviour

## Screenshots (if applicable)
![earn_opp_1](https://user-images.githubusercontent.com/5720927/162684358-edf3178c-d393-4fc7-b26a-c0d47cfa5cb6.png)

![opp_cards_1](https://user-images.githubusercontent.com/5720927/162683672-1af59af9-aee6-47d9-abfa-dbb0dfa765b6.png)

![earn_opp_2](https://user-images.githubusercontent.com/5720927/162684378-0f52b948-5de4-45c0-8766-c7e6b62dcc83.png)
